node/testdata/setup.ts

```diff
@@ -0,0 +1,177 @@
+      if (response.ok) {
+        if (!response.body) {
+          throw new Error(
+            `The requested download url ${url} doesn't contain an archive to download`,
+          );
+        }
+        return response.body.getIterator();
+      } else if (response.status === 404) {
+        throw new Error(
+          `The requested version ${config.nodeVersion} could not be found for download`,
+        );
+      }
+      throw new Error(`Request failed with status ${response.status}`);
+    });
+
+  const filePath = fromFileUrl(new URL(path, import.meta.url));
+
+  await ensureFile(filePath);
+  const file = await Deno.open(filePath, {
+    truncate: true,
+    write: true,
+  });
+  for await (const chunk of fileContent) {
+    await Deno.write(file.rid, chunk);
+  }
+  file.close();
+  console.log(`Downloaded: ${url} into ${path}`);
+}
+
+async function clearTests() {
+  console.log("Cleaning up previous tests");
+
+  const files = walk(
+    (fromFileUrl(new URL(config.suitesFolder, import.meta.url))),
+    {
+      includeDirs: false,
+      skip: ignoreList,
+    },
+  );
+
+  for await (const file of files) {
+    await Deno.remove(file.path);
+  }
+}
+
+/**
+ * This will iterate over the ignore and test lists defined in the
+ * configuration file
+ * 
+ * If it were to be found in the ignore list or not found in the test list, the
+ * function will return undefined, meaning the file won't be regenerated
+ */
+function getRequestedFileSuite(file: string): string | undefined {
+  for (const regex of ignoreList) {
+    if (regex.test(file)) {
+      return;
+    }
+  }
+
+  for (const suite in config.tests) {
+    for (const regex of config.tests[suite]) {
+      if (new RegExp(regex).test(file)) {
+        return suite;
+      }
+    }
+  }
+}
+
+async function decompressTests(filePath: string) {
+  console.log(`Processing ${basename(filePath)}...`);
+  const compressedFile = await Deno.open(
+    new URL(filePath, import.meta.url),
+    { read: true },
+  );
+
+  const buffer = new Deno.Buffer(gunzip(await Deno.readAll(compressedFile)));
+  Deno.close(compressedFile.rid);
+
+  const tar = new Untar(buffer);
+
+  for await (const entry of tar) {
+    if (entry.type !== "file") continue;
+    const suite = getRequestedFileSuite(entry.fileName);
+    if (!suite) continue;
+    const path = resolve(
+      fromFileUrl(new URL(config.suitesFolder, import.meta.url)),
+      suite,
+      basename(entry.fileName),
+    );
+    await ensureFile(path);
+    const file = await Deno.open(path, {
+      create: true,
+      truncate: true,
+      write: true,
+    });
+    // This will allow CI to pass without checking linting and formatting
+    // on the test suite files, removing the need to mantain that as well
+    await Deno.writeAll(
+      file,
+      new TextEncoder().encode(
+        "// deno-fmt-ignore-file\n// deno-lint-ignore-file\n" +
+          "\n// Copyright Joyent and Node contributors. All rights reserved. MIT license.\n\n",
```

Comment:
@Soremwar How about adding version numbers of Node.js around these lines to prevent the confusion about versions? like:

```js
// Copied from Node.js v15.5.1
```

I also think it would be good to add note that it's a generated file.

```js
// This files is automatically generated by ./node/testdata/setup.ts. Do not modify this file manually.
```

Reply:
I like this. Just added it